src/Internal/Service/Level3Service.php

```diff
@@ -112,6 +104,29 @@ public function get_data_from_order( int $order_id ): array {
 			$level3_data['shipping_from_zip'] = $store_postcode;
 		}
 
+		/**
+		 * Filters the Level 3 data based on order.
+		 *
+		 * Example usage: Enables updating the discount based on the products in the order,
+		 * if any of the products are gift cards.
+		 *
+		 * @since 7.9.3
+		 *
+		 * @param array $level3_data Precalculated Level 3 data based on order.
+		 * @param WC_Order $order    The order object.
+		 */
+		$level3_data = apply_filters( 'wcpay_payment_request_level3_data', $level3_data, $order );
+
+		if ( count( $level3_data['line_items'] ) > 200 ) {
+			// If more than 200 items are present, bundle the last ones in a single item.
+			$items_to_send = array_merge(
+				array_slice( $level3_data['line_items'], 0, 199 ),
+				[ $this->bundle_level3_data_from_items( array_slice( $level3_data['line_items'], 200 ) ) ]
```

Comment:
```suggestion
				[ $this->bundle_level3_data_from_items( array_slice( $level3_data['line_items'], 199 ) ) ]
```
This wasn't directly your contribution, but shouldn't the second `array_slice` begin at the offset of 199? Doesn't this skip out `$level3_data['line_items'][199]` otherwise or am I miscalculating? 🤔

Reply:
Good observation! You're right.